entity-framework/core/managing-schemas/migrations/managing.md

```diff
@@ -0,0 +1,199 @@
+---
+title: Managing Migrations - EF Core
+author: bricelam
+ms.author: bricelam
+ms.date: 05/06/2020
+uid: core/managing-schemas/migrations/managing
+---
+# Managing Migrations
+
+As your model changes, migrations are added and removed as part of normal development, and the migration files are checked into your project's source control. To manage migrations, you must first install the [EF Core command-line tools](xref:core/miscellaneous/cli/index).
+
+> [!TIP]
+> If the `DbContext` is in a different assembly than the startup project, you can explicitly specify the target and startup projects in either the [Package Manager Console tools](xref:core/miscellaneous/cli/powershell#target-and-startup-project) or the [.NET Core CLI tools](xref:core/miscellaneous/cli/dotnet#target-project-and-startup-project).
+
+## Add a migration
+
+After your model has been changed, you can add a migration for that change:
+
+### [.NET Core CLI](#tab/dotnet-core-cli)
+
+```dotnetcli
+dotnet ef migrations add AddBlogCreatedTimestamp
+```
+
+### [Visual Studio](#tab/vs)
+
+``` powershell
+Add-Migration AddBlogCreatedTimestamp
+```
+
+***
+
+The migration name can be used like a commit message in a version control system. For example, you might choose a name like *AddBlogCreatedTimestamp* if the change is a new `CreatedTimestamp` property on your `Blog` entity.
+
+Three files are added to your project under the **Migrations** directory:
+
+* **XXXXXXXXXXXXXX_AddCreatedTimestamp.cs**--The main migrations file. Contains the operations necessary to apply the migration (in `Up`) and to revert it (in `Down`).
+* **XXXXXXXXXXXXXX_AddCreatedTimestamp.Designer.cs**--The migrations metadata file. Contains information used by EF.
+* **MyContextModelSnapshot.cs**--A snapshot of your current model. Used to determine what changed when adding the next migration.
+
+The timestamp in the filename helps keep them ordered chronologically so you can see the progression of changes.
+
+### Namespaces
+
+You are free to move Migrations files and change their namespace manually. New migrations are created as siblings of the last migration. Alternatively, you can specify the namespace at generation time as follows:
+
+### [.NET Core CLI](#tab/dotnet-core-cli)
+
+```dotnetcli
+dotnet ef migrations add InitialCreate --namespace Your.Namespace
+```
+
+### [Visual Studio](#tab/vs)
+
+``` powershell
+Add-Migration InitialCreate -Namespace Your.Namespace
+```
+
+***
+
+## Customize migration code
+
+While EF Core generally creates accurate migrations, you should always review the code and make sure it corresponds to the desired change; in some cases, it is even necessary to do so.
+
+### Column renames
+
+One notable examine where customizing migrations is required is when renaming a property. For example, if you rename a property from `Name` to `FullName`, EF Core will generate the following migration:
```

Comment:
```suggestion
One notable example where customizing migrations is required is when renaming a property. For example, if you rename a property from `Name` to `FullName`, EF Core will generate the following migration:
```

But I don't think this is actually true--we usually detect property renames. We're a little worse about renaming entity types.